README.md

```diff
@@ -123,35 +123,50 @@ If you are using the `docker/build-push-action`, and would like to pass the SSH
             default=${{ env.SSH_AUTH_SOCK }}
 ```
 
-### Using the `docker/build-push-action` Action together with multiple Deploy Keys
+Make sure not to miss the next section, though.
 
-If you use the `docker/build-push-action` and want to use multiple GitHub deploy keys, you need to copy the git and ssh configuration to the container during the build. Otherwise, the Docker build process would still not know how to handle multiple deploy keys. Even if the ssh agent was set up correctly on the runner.
+### Forwarding the SSH agent into Docker build processes, together with multiple Deploy Keys
 
-This requires an additional step in the actions workflow and two additional lines in the Dockerfile.
+If you use one of:
+* the `docker/build-push-action`
+* manual `docker build`
+* manual `docker compose build` 
+
+_and_ want to use multiple GitHub deploy keys, you need to copy the Git and SSH configuration to the container during the build. This is necessary so that Git can pick the right one from your deployment keys. This is necessary _in addition to_ forwarding the SSH agent socket into the build process.
+
+This requires an additional step in the workflow file **after** the `ssh-agent` step and **before** the Docker build step.  You also need two additional lines in the `Dockerfile` to actually copy the configs.
+
+The following example will: 
+* collect the necessary Git and SSH configuration files in a directory that must be part of the Docker build context so that... 
+* ... the files can be copied into the Docker image (or an intermediate build stage).
 
 Workflow:
+
 ```yml
-      - name: Prepare git and ssh config for build context
+      - name: ssh-agent setup
+        ...
+
+      - name: Collect Git and SSH config files in a directory that is part of the Docker build context
         run: |
           mkdir root-config
           cp -r ~/.gitconfig  ~/.ssh root-config/
   
-      - name: Build and push
-        id: docker_build
-        uses: docker/build-push-action@v2
-        with:
-          ssh: |
-            default=${{ env.SSH_AUTH_SOCK }}
+      - name: Docker build 
+        # build-push-action | docker [compose] build | etc.
```

Comment:
@j-riebe should we mention here that additional flags/parameters need to be given to `docker` to forward the `SSH_AUTH_SOCK`?

Reply:
I think a hint would be good (although it should be "clear" from the previous section).

Would you only mention it (and expect the user to be aware of the `--ssh` parameter) or provide examples?
Like:
* build-push-action (copied from section above)
```yml
      - name: Build and push
        id: docker_build
        uses: docker/build-push-action@v2
        with:
          ssh: |
            default=${{ env.SSH_AUTH_SOCK }}
```
* ```docker build --ssh default=SSH_AUTH_SOCK``` (docker docs)
* ```docker compose build --ssh default=SSH_AUTH_SOCK``` (docker docs)

Maybe we just append the `--ssh default=${{ env.SSH_AUTH_SOCK}}` to the lines 132 & 133.
That should make it pretty obvious.